### DISCUSSION TITLE
Bundling init scripts to addons

### DISCUSSION THREAD

**Discussion by petervelosy** [freeplane/freeplane#1041] [Help]
Dear Community,

I am developing an addon in which I would like to attach a listener to every node deletion event. Currently, the only way I've found to do this is by means of an init script. 

However, init scripts are not considered by Freeplane to be a part of an addon, as they do not live in the addon's own script directory. Consequently, even if I can do a workaround for having an addon install an init script (by adding the init script to "zips"), Freeplane's security warning pops up the first time the init script gets executed (i.e. like if the init script were something new from an untrusted source instead of being a trusted part of an addon). This is not very user friendly.

Can you recommend me a good workaround here?

Many thanks!
Peter

**Comment by lilive** [comment]
Hello,
I did the same thing for my [bookmarks addon](https://github.com/lilive/Freeplane-Bookmarks-add-on). I don't remember about Freeplane security warning. Perhaps because I set "script execution enabled" to "Yes" in the preferences ? Not sure at all, but I see that I recommend to users to do the same in the readme.md
Indeed, an addon init scripts feature would be nice.
